src/modules/meson.rs

```diff
@@ -0,0 +1,232 @@
+                _ => None,
+            })
+            .parse(None, Some(context))
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `meson`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn get_first_grapheme(text: &str) -> &str {
+    UnicodeSegmentation::graphemes(text, true)
+        .next()
+        .unwrap_or("")
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::test::ModuleRenderer;
+    use nu_ansi_term::Color;
+
+    #[test]
+    fn test_changed_truncation_symbol() {
+        test_truncate_length_with_config(
+            "1337_hello_world",
+            15,
+            "1337_hello_worl",
+            "%",
+            "truncation_symbol = \"%\"",
+        )
+    }
+
+    #[test]
+    fn test_no_truncation_symbol() {
+        test_truncate_length_with_config(
+            "1337_hello_world",
+            15,
+            "1337_hello_worl",
+            "",
+            "truncation_symbol = \"\"",
+        )
+    }
+
+    #[test]
+    fn test_multi_char_truncation_symbol() {
+        test_truncate_length_with_config(
+            "1337_hello_world",
+            15,
+            "1337_hello_worl",
+            "a",
+            "truncation_symbol = \"apple\"",
+        )
+    }
+
+    #[test]
+    fn test_ascii_boundary_below() {
+        test_truncate_length("1337_hello_world", 15, "1337_hello_worl", "…")
+    }
+
+    #[test]
+    fn test_ascii_boundary_on() {
+        test_truncate_length("1337_hello_world", 16, "1337_hello_world", "")
+    }
+
+    #[test]
+    fn test_ascii_boundary_above() {
+        test_truncate_length("1337_hello_world", 17, "1337_hello_world", "")
+    }
+
+    #[test]
+    fn test_one() {
+        test_truncate_length("1337_hello_world", 1, "1", "…")
+    }
+
+    #[test]
+    fn test_zero() {
+        test_truncate_length("1337_hello_world", 0, "1337_hello_world", "")
+    }
+
+    #[test]
+    fn test_negative() {
+        test_truncate_length("1337_hello_world", -1, "1337_hello_world", "")
+    }
+
+    #[test]
+    fn test_hindi_truncation() {
+        test_truncate_length("नमस्ते", 3, "नमस्", "…")
+    }
+
+    #[test]
+    fn test_hindi_truncation2() {
+        test_truncate_length("नमस्त", 3, "नमस्", "…")
+    }
+
+    #[test]
+    fn test_japanese_truncation() {
+        test_truncate_length("がんばってね", 4, "がんばっ", "…")
+    }
+
+    #[test]
+    fn not_in_env() {
+        let actual = ModuleRenderer::new("meson").collect();
+
+        let expected = None;
+
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn env_set() {
+        let actual = ModuleRenderer::new("meson")
+            .env("MESON_DEVENV", "1")
+            .env("MESON_PROJECT_NAME", "starship")
+            .collect();
+
+        let expected = Some(format!(
+            "in {} ",
+            // TODO change the icon and the color
+            Color::Blue.bold().paint("⬢ starship")
```

Comment:
Does this still need to be handled?
```suggestion
            Color::Blue.bold().paint("⬢ starship")
```